kombu/transport/redis.py

```diff
@@ -944,6 +944,7 @@ def _connparams(self, asynchronous=False):
             'host': conninfo.hostname or '127.0.0.1',
             'port': conninfo.port or self.connection.default_port,
             'virtual_host': conninfo.virtual_host,
+            'username': conninfo.userid,
```

Comment:
should we also update the related docs?

Reply:
@auvipy Actually, this is a good question. I'd say yes if the documentation explicitly says that the username part of the connection URL won't be parsed (before this change), otherwise, I'd say that this should be "mainstream" that a connection URL can contain a username.

If you would like me, I'm happy to find the relevant section in the documentation and explicitly state that, from now on, usernames will be parsed too.

Reply:
Please do.

Reply:
@thedrow @auvipy I'm happy to extend any documentation, if you could advise a bit. The documentation has no mention at all about usernames in terms of the Redis transport. In fact, the documentation is defined as
```rst
.. code-block::

    redis://REDIS_ADDRESS[:PORT][/VIRTUALHOST]
    rediss://REDIS_ADDRESS[:PORT][/VIRTUALHOST]

To use sentinel for dynamic Redis discovery,
the connection string has the following format:

.. code-block::

    sentinel://SENTINEL_ADDRESS[:PORT]
```
I do believe -- but correct me if I'm wrong -- the reason behind this is that the host parameter (`REDIS_ADDRESS`) may contain username and password as per RFC 2617. If you would like me, I can describe that in the inline [documentation](https://github.com/celery/kombu/blob/master/kombu/transport/redis.py#L12-L27), though that would be a bit verbose.

If you would like me to extend another documentation that I did not find (yet), please help me with the link.

Reply:
The documentation is not 100% complete. I am not 100% whether is good idea to refer to some standard because URL parsing is done by `urllib.parse.urlparse` but some magic is involved during the parsing - see `kombu/utils/url.py:url_to_parts()`. Let it keep simple. Please extend the documentation as follows:

```
redis://[USER:PASSWORD@]REDIS_ADDRESS[:PORT][/VIRTUALHOST]
rediss://[USER:PASSWORD@]REDIS_ADDRESS[:PORT][/VIRTUALHOST]
```

Maybe we can also add some examples to the documentation.

Otherwise, the PR seems to be fine for me.